conda_smithy/lint_recipe.py

```diff
@@ -224,10 +242,17 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
         if not expected_subsections:
             continue
         for subsection in get_section(meta, section, lints):
-            if subsection not in expected_subsections:
+            if section != 'source' and subsection not in expected_subsections:
                 lints.append('The {} section contained an unexpected '
                              'subsection name. {} is not a valid subsection'
                              ' name.'.format(section, subsection))
+            elif section == 'source':
+                for source_subsection in subsection:
+                    if source_subsection not in expected_subsections:
+                        lints.append('The {} section contained an unexpected '
+                                     'subsection name. {} is not a valid subsection'
+                                     ' name.'.format(section, source_subsection))
+
```

Comment:
This is not very elegant, but it was the best I could come up while trying to minimize the number of lines modified.

Reply:
eh, it works :)